### PR TITLE
wrap "rm -rf" with docker

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by MHBauer** [comment]
I'd prefer to try #1729

**Comment by MHBauer** [comment]
This is also a ridiculous solution. That said, I'm all for "if it's stupid and it works it's not stupid", but we can do better.

**Comment by duglin** [comment]
#1729 solves a different problem

**Comment by MHBauer** [comment]
What does this solve?

**Comment by duglin** [comment]
fixes the "make clean" failure when run locally too

**Comment by MHBauer** [comment]
Then put it on every rm

**Comment by duglin** [comment]
we only need it on docker create output. If there are others then yes we should add it there too. Just let me know which other "rm"s fail

**Comment by MHBauer** [comment]
Shouldn't it also be on the generated files and the go tooling output?

**Comment by duglin** [comment]
@MHBauer done.
What's funny is that `make clean` will erase the .scBuildImage which is needed by some of the rm's. So doing `make clean` multiple times will actually recreate the image, and then delete it. :-)

**Comment by MHBauer** [comment]
That's exactly the problem we were encountering. Maybe we can solve this by judicious ordering of make rules, but even so, calling any individual one has that dependency.

**Comment by kibbles-n-bytes** [comment]
@arschles Check the discussion at #1713 . The files that we build inside docker are owned by root, and makes `rm -rf bin` (and thus `make clean`) impossible without running as root on your machine. Running it in docker is a quick hack to get it to work correctly, while long term we need to fix the permissions. A bug we hit when releasing `v0.1.6`, though, is that running `make clean` when you don't have `scbuildimage` fails, so we also need to add that as a prereq.

We tried getting rid of docker and just living with the root requirement, but it broke the `v0.1.7` release, as Travis apparently doesn't run as root.

**Comment by kibbles-n-bytes** [comment]
As this is the quickest and easiest thing to get our releases (hopefully) working again, I am happy with this going in. I tested it on my local machine and seems to work fine.

**Comment by kibbles-n-bytes** [comment]
#1729 is now merged (@MHBauer strong-armed me lol), so this should be mostly unnecessary now.

**Comment by duglin** [comment]
I'm not following. Are we no longer seeing an issue with "make clean*" failing to erase files that were created while running under docker?

**Comment by kibbles-n-bytes** [comment]
@duglin We are still seeing the issue. But the release is no longer blocked, which means it's less of a priority to fix it immediately. This is a quick-n-easy fix, though long-term I think we should fix the permissions on the resources coming out of docker. Just depends how much we're hurting by having to `sudo make clean`. I'm happy either way, though, so I'll keep my LGTM here.

File: Makefile [hunk 1]
```diff
@@ -265,12 +265,12 @@ test-e2e: build-e2e
 
 clean: clean-bin clean-build-image clean-generated clean-coverage
 
-clean-bin:
-	rm -rf $(BINDIR)
+clean-bin: .init $(scBuildImageTarget)
+	$(DOCKER_CMD) rm -rf $(BINDIR)
 	rm -f .generate_exes
 
 clean-build-image:
```
Comment (MHBauer) [inline Makefile]: does this need init and stuff as well?

Comment (duglin) [reply Makefile]: yep - fixed

**Comment by MHBauer** [comment]
I think that'll do it. Feel free to merge when it builds. Though, I doubt it'll fail.

**Pull request merged**